In [1]:
from scipy.stats import norm

In [5]:
import numpy as np

In [7]:
(1850-1800)/(100/np.sqrt(50))

3.5355339059327378

In [2]:
1-norm.cdf(3.53)

0.0002077798334806369

In [3]:
1-norm.cdf(1.19)

0.11702319602310873

In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

from scipy.stats import norm

## ...

In [3]:
df = pd.read_csv("Sachin_ODI.csv")

In [12]:
df.head()

runs  NotOut mins  bf  fours  sixes      sr  Inns           Opp    Ground  \
0    13       0   30  15      3      0   86.66     1   New Zealand    Napier   
1    37       0   75  51      3      1   72.54     2  South Africa  Hamilton   
2    47       0   65  40      7      0  117.50     2     Australia   Dunedin   
3    48       0   37  30      9      1  160.00     2    Bangladesh   Sharjah   
4     4       0   13   9      1      0   44.44     2      Pakistan   Sharjah   

         Date        Winner    Won  century  
0  1995-02-16   New Zealand  False    False  
1  1995-02-18  South Africa  False    False  
2  1995-02-22         India   True    False  
3  1995-04-05         India   True    False  
4  1995-04-07      Pakistan  False    False

In [4]:
df.shape

(360, 14)

## .

In [15]:
mu = df["runs"].mean()
mu

43.24166666666667

In [16]:
sigma = df["runs"].std()
sigma

42.18245206052442

## First Inning

H0 : avg runs in first innning = 43.24 <br>
HA : avg runs in first innning > 43.24

In [17]:
first_innings_runs = df[df["Inns"]==1][["runs"]]

In [18]:
first_innings_runs

runs
0      13
7      30
15    137
16      3
17     31
..    ...
345   111
346     2
348    85
357     6
358   114

[170 rows x 1 columns]

In [19]:
len(first_innings_runs) ## n = 170

170

In [20]:
data = first_innings_runs

In [21]:
data.mean() # test statistic

runs    46.670588
dtype: float64

In [22]:
mu

43.24166666666667

In [23]:
z_stat = (data.mean() - mu)/(sigma/np.sqrt(len(data)))
z_stat

runs    1.059864
dtype: float64

In [25]:
p_value = 1 - norm.cdf(z_stat)
p_value

array([0.14460321])

In [26]:
alpha = 0.15
if p_value < alpha:
    print("Reject Null Hypotheis")
else:
    print("Stick to the null hypothesis")

Reject Null Hypotheis


## When Aus. won against India

H0 : avg runs = 43.24 <br>
HA : avg run < 43.24

In [29]:
data = df[df.Winner=="Australia"][["runs"]]

In [30]:
len(data)

38

In [31]:
data.mean()

runs    33.973684
dtype: float64

In [32]:
z_stat = (data.mean() - mu)/(sigma/np.sqrt(len(data)))
z_stat

runs   -1.354394
dtype: float64

In [32]:
p_value = norm.cdf(z_stat)

In [33]:
p_value

array([0.08780528])

In [36]:
alpha = 0.05 
if p_value < alpha:
    print("Reject Null Hypotheis")
else:
    print("Stick to the null hypothesis")

Stick to the null hypothesis


 ## Scoring pattern for matches won

H0: average runs in matches won = 43.2 <br>
HA: average runs in matches won > 43.2